src/js/jsx/sections/style/AppearancePanelSections.jsx

```diff
@@ -0,0 +1,491 @@
+                            title: style,
+                            style: {
+                                "fontFamily": familyName,
+                                "fontStyle": this._getCSSFontStyle(searchableStyle),
+                                "fontWeight": this._getCSSFontWeight(searchableStyle)
+                            }
+                        };
+                    }, this)
+                    .toList();
+            }
+
+            if (textLayers.size > 0 && vectorLayers.size === 0) {
+                overlay = function (colorTiny) {
+                    var fillStyle = {
+                        height: "100%",
+                        width: "100%",
+                        backgroundColor: colorTiny ? colorTiny.toRgbString() : "transparent"
+                    };
+                    if (familyName === nls.localize("strings.STYLE.TYPE.MIXED")) {
+                        return (
+                            <div
+                                className="fill__preview"
+                                style={fillStyle}/>
+                        );
+                    } else {
+                        var typeStyle = {
+                            fontFamily: familyName || "helvetica",
+                            fontStyle: this._getCSSFontStyle(styleTitle) || "regular",
+                            fontWeight: this._getCSSFontWeight(styleTitle) || 400,
+                            fontSize: 24
+                        };
+
+                        if (colorTiny) {
+                            typeStyle.color = colorTiny.toRgbString();
+                        }
+
+                        return (
+                            <div
+                                className="type-vector-color__preview"
+                                style={typeStyle}>
+                                Aa
+                            </div>
+                        );
+                    }
+                }.bind(this);
+            } else {
+                overlay = function (colorTiny) {
+                    var fillStyle = {
+                        height: "100%",
+                        width: "100%",
+                        backgroundColor: colorTiny ? colorTiny.toRgbString() : "transparent"
+                    };
+                    return (
+                        <div
+                            className="type-vector-color__preview"
+                            style={fillStyle}/>
+                    );
+                };
+
+                if (textLayers.size === 0 && vectorLayers.size > 0) {
+                    colors = this.state.vectorFill.colors;
+                } else {
+                    if (characterStyles && !characterStyles.isEmpty()) {
+                        var textColors = collection.pluck(characterStyles, "color");
+                        var vectorColors = this.state.vectorFill.colors;
+                        var temp = textColors.concat(vectorColors);
```

Comment:
@iwehrman Can you help me here? The problem that happens is that `textColors` is in the same format as `vectorColors` but apparently the `r, g, b, a,` values of `textColors` is not rounded to the nearest integer and `vectorColors` are. Should I just call `Math.round()` on each of these values or am I missing something if I do that? Just not sure because we clearly handle text colors and vector fills differently so I wanted to double check. @pineapplespatula You worked with type right? So maybe you know better? 

The end goal of this is getting all the vector fills and all the text colors, and putting them into one final data structure that is of the same type as `textColors`.